examples/api-extension/runner/src/main.rs

```diff
@@ -0,0 +1,101 @@
+/* Copyright (c) Fortanix, Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+extern crate aesm_client;
+extern crate enclave_runner;
+extern crate sgxs_loaders;
+extern crate failure;
+#[macro_use]
+extern crate clap;
+
+use aesm_client::AesmClient;
+use enclave_runner::{ApiExtension, EnclaveBuilder, OutputBuffer, SyncStream};
+use failure::{Error, ResultExt};
+use sgxs_loaders::isgx::Device as IsgxDevice;
+
+use clap::{App, Arg};
+use std::fs::{File, OpenOptions};
+use std::io::{ErrorKind as IoErrorKind, Result as IoResult};
+
+arg_enum!{
+    #[derive(PartialEq, Debug)]
+    #[allow(non_camel_case_types)]
+    pub enum Signature {
+        coresident,
+        dummy
+    }
+}
+
+struct FileStream {
```

Comment:
A file is a poor example because it doesn't really resemble a bi-directional stream.